### ISSUE TITLE
[BUG] win_domain_group_membership module should fail on missing ActiveDirectory module

### ISSUE THREAD

**Issue by scottd018** [ansible/ansible#57411]
##### SUMMARY

When running the win_domain_group_membership module on a Windows system which does not have the Active Directory PowerShell module available, executing of the playbook will continue and eventually fail with error:

    "msg": "Could not find domain user, group, service account or computer named HOSTNAME$"

Upon further investigation, the ActiveDirectory tools module is not installed on the Windows Server.  Installing the ActiveDirectory tools fixes the problem.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

win_domain_group_membership

##### ANSIBLE VERSION

```
ansible 2.8.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/var/lib/awx/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Mar 26 2019, 22:13:06) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
```

##### CONFIGURATION

```
-bash-4.2$ ansible-config dump --only-changed
-bash-4.2$ 
```

##### OS / ENVIRONMENT

Target OS Versions = *Windows*

##### STEPS TO REPRODUCE

Ensure that the Windows host is missing the ActiveDirectory module and run the playbook.

```
- name: "add ad_computer_object_name: [{{ ad_computer_object_name }}] to ad_security_group: [{{ ad_security_group }}]"
  win_domain_group_membership:
    name: "{{ ad_security_group }}"
    members:
      - "{{ ad_computer_object_name }}$"
    state: 'present'
```

##### EXPECTED RESULTS

Based on lines 8-13 in the module, I would expect it to fail with error: "win_domain_group_membership requires the ActiveDirectory PS module to be installed"

Lines 8-13:

```
try {
    Import-Module ActiveDirectory
}
catch {
    Fail-Json -obj @{} -message "win_domain_group_membership requires the ActiveDirectory PS module to be installed"
}
```

##### ACTUAL RESULTS

Fails on lines 57-60 in the module:

```
    $group_member = Get-ADObject -Filter "SamAccountName -eq '$member' -and $ad_object_class_filter" -Properties objectSid, sAMAccountName
    if (!$group_member) {
        Fail-Json -obj $result "Could not find domain user, group, service account or computer named $member"
    }
```

```
{
    "_ansible_parsed": true,
    "added": [],
    "_ansible_no_log": false,
    "_ansible_delegated_vars": {
        "ansible_delegated_host": "HOSTNAME",
        "ansible_host": "HOSTNAME"
    },
    "changed": false,
    "msg": "Could not find domain user, group, service account or computer named HOSTNAME$",
    "removed": []
}
```

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/windows/win_domain_group_membership.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_domain_group_membership.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @ShachafGoldstein @if-meaton @jborean93 @jhawkesworth @jiuka @nitzmahone
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ShachafGoldstein** [comment]
What version of PS and Windows are you running on?

**Comment by scottd018** [comment]
@ShachafGoldstein PowerShell v5 and Windows Server 2016 (also tested with the same results on Server 2019 - no output to show for that):

```
            "ansible_kernel": "10.0.14393.0",
            "ansible_powershell_version": 5,
```

**Comment by ShachafGoldstein** [comment]
OK, can you open a console on one of the server that don't have the feature
installed and type the import command and replay here whether you get an
error result out or an empty prompt?

**Comment by scottd018** [comment]
@ShachafGoldstein Here is the output of the failure to `Import-Module ActiveDirectory`:

```
PS C:\Windows\system32> import-module activedirectory

import-module : The specified module 'activedirectory' was not loaded because no valid module file was found in any

module directory.

At line:1 char:1

+ import-module activedirectory

+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    + CategoryInfo          : ResourceUnavailable: (activedirectory:String) [Import-Module], FileNotFoundException

    + FullyQualifiedErrorId : Modules_ModuleNotFound,Microsoft.PowerShell.Commands.ImportModuleCommand
```

Also, I told you wrong in the last comment.  This was Windows Server 2012 R2 with PowerShell 4 (was running my setup against the wrong server).  Apologies for the confusion:

```
            "ansible_kernel": "6.3.9600.0",
            "ansible_powershell_version": 4,
```

**Comment by ShachafGoldstein** [comment]
I'm pretty sure the problem is that `erroractionpreference` isn't set to stop.
mind adding `$ErrorActionPreference = 'stop'` in line 7 and test to make sure?

**Comment by scottd018** [comment]
@ShachafGoldstein that did the trick:

With `$ErrorActionPreference = 'stop'`:

```
Using module file /tmp/mt_bank/library/win_domain_group_membership.ps1
Pipelining is enabled.
<HOST> ESTABLISH WINRM CONNECTION FOR USER: <USER> on PORT 5986 TO <HOST>
EXEC (via pipeline wrapper)
fatal: [<HOST>]: FAILED! => {
    "changed": false, 
    "msg": "win_domain_group_membership requires the ActiveDirectory PS module to be installed"
}
```

**Comment by ShachafGoldstein** [comment]
I'll write a PR for this later

**Comment by jborean93** [comment]
This shouldn't be necessary, since Ansible 2.8 we always set `$ErrorActionPreference = 'Stop'` [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/powershell/module_wrapper.ps1#L46) when executing our scripts. Also the fact that it didn't error out when actually using the `Get-ADObject` cmdlet means that the cmdlet was imported in some shape or form.

Something weird is going on here that we need to track down a bit more. Hopefully I'll have some spare time to figure out why our EAP preference isn't applying ehre.

**Comment by ShachafGoldstein** [comment]
I think it only errored on get since it returned null since it was a non existing cmdlet, and bot the import and get gave errors that were simply ignored

@scottd018 can you verify you are using 2.8 as you wrote in the initial description?

**Comment by scottd018** [comment]
@jborean93 @ShachafGoldstein hey guys, you all are correct. I just looked again, and it was user error. We had backported the change via a library in Ansible 2.7, which failed with the error, but just verified this worked as expected in Ansible 2.8. Because of that, I think the code is working as intended.

Sorry for the confusion. This was a case of too many windows open and I grabbed the output from the wrong version. Womp womp...
